R/designOfExperiments.R

```diff
@@ -17,4 +17,188 @@
+                               "taguchi" = gettext("Available Taguchi Designs"))
+
+    table <- createJaspTable(tableTitle)
+    table$position <- position
+
+    table$dependOn(options = "design")
+
+    if(options[["design"]] == "factorial"){
+
+      table$addColumnInfo(name = 'run', 		title = gettext("Runs"), 	type = 'integer')
+      table$addColumnInfo(name = 'factor2', 	title = "2", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor3', 	title = "3", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor4', 	title = "4", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor5', 	title = "5", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor6', 	title = "6", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor7', 	title = "7", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor8', 	title = "8", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor9', 	title = "9", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor10', 	title = "10", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor11', 	title = "11", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor12', 	title = "12", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor13', 	title = "13", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor14', 	title = "14", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor15', 	title = "15", 				type = 'string', overtitle = "Factors")
+
+      rows <- data.frame(run = c(4, 8, 16, 32, 64, 128),
+                         factor2 = c("Full", "", "", "", "", ""),
+                         factor3 = c("III", "Full", "", "", "", ""),
+                         factor4 = c("", "IV", "Full", "", "", ""),
+                         factor5 = c("", "III", "V", "Full", "", ""),
+                         factor6 = c("", "III", "IV", "VI", "Full", ""),
+                         factor7 = c("", "III", "IV", "IV", "VII", "Full"),
+                         factor8 = c("", "", "IV", "IV", "V", "VIII"),
+                         factor9 = c("", "", "III", "IV", "V", "VI"),
+                         factor10 = c("", "", "III", "IV", "V", "V"),
+                         factor11 = c("", "", "III", "IV", "V", "V"),
+                         factor12 = c("", "", "III", "IV", "IV", "IV"),
+                         factor13 = c("", "", "III", "IV", "IV", "IV"),
+                         factor14 = c("", "", "III", "IV", "IV", "IV"),
+                         factor15 = c("", "", "III", "IV", "IV", "IV"))
+
+      jaspResults[["state"]] <- createJaspState(rows)
+      jaspResults[["state"]]$dependOn("always_present")
```

Comment:
Does this actually work? I assume this is a way to make sure that this table never needs to be recomputed? Does Joris know about this trick? :p

Reply:
I think he might, the help file in R for createJaspState says that if a state has no dependencies it will always be thrown away. I guess that means that when there is a dependency that never changes it never gets thrown away? :p

Reply:
@JorisGoosen Does this work?